index.js

```diff
@@ -39,6 +56,31 @@ app.get("/messages", (req, res) => {
   res.json(messages.slice(afterId));
 });
 
+app.put("/message/:id", (req, res) => {
+  if (!messages[req.params.id - 1]) {
+    return res.status(404).json({ error: "Unknown item" });
+  }
+  if (req.body.message) {
+    messages[req.params.id - 1].message = req.body.message;
+  }
+  if (req.body.user) {
+    messages[req.params.id - 1].user = req.body.user;
+  }
+  res.json(messages[req.params.id - 1]);
+});
+
+app.delete("/message/:id", (req, res) => {
+  // doesn't work
+  if (!messages[req.params.id - 1]) {
+    return res.status(404).json({ error: "Unknown item" });
+  }
+  const index = messages.indexOf(req.body.message);
```

Comment:
`messages` is an array of objects. I'm not sure what `req.body.message` is here. Instead you could use messages.findIndex(...)` to find the index of the message that has the specified id (`req.params.id`).

Reply:
I haven't working on fixing the delete endpoint yet